torch/_dynamo/source.py

```diff
@@ -560,6 +560,17 @@ def reconstruct(self, codegen):
         codegen.extend_output(create_call_function(1, True))
 
 
+# NB: We don't expect you to actually ever generate guards against this
+# source, it is ephemeral
```

Comment:
Could we throw somewhere inside the class to make sure this is the case?

Reply:
I think this is inadvisable. The only place we can throw the assertion is in name(), but for debugging purposes you want to always be able to query the name of a source.